### PR TITLE
Updating Spin tutorials for 2.0 release

### PR DESCRIPTION
**Pull request by mikkelhegn**
**Bart check**

tpmccallum@192-168-1-22 developer % bart check content/spin/v2/ai-sentiment-analysis-api-tutorial.md 
✅ content/spin/v2/ai-sentiment-analysis-api-tutorial.md
tpmccallum@192-168-1-22 developer % bart check content/spin/v2/key-value-store-tutorial.md
✅ content/spin/v2/key-value-store-tutorial.md
tpmccallum@192-168-1-22 developer % bart check content/spin/v2/serverless-ai-hello-world.md 
✅ content/spin/v2/serverless-ai-hello-world.md
tpmccallum@192-168-1-22 developer % bart check content/spin/v2/template-authoring.md
✅ content/spin/v2/template-authoring.md

**npm run test**

Screenshot 2023-11-01 at 13 51 59

Closing #957
Closing #958

This is the PR meant to adress the above two issues. Current a rough first draft, but will keep updating as I get more done.

### CODE REVIEW

**Comment by tpmccallum** [comment]
Thanks @mikkelhegn 

Please ping me for a review/approval as soon as you are ready.
We are continually/incrementally merging any updated pages in the new `v2` folder (all of which have their frontmatter’s date set to the release date). 
This is a great PR, thanks. I like the way you have only made changes to `v2` files in one PR (allowing us to approve/merge as we go). Great stuff!

**Comment by tpmccallum** [comment]
Essentially, as soon as we can merge new content the better because it removes any chance of line conflicts in GitHub. 
We are treating `v2` like a live site - changes coming in hot - (except that `v2` is actually invisible until we flip the config and make pages and their corresponding entries in the sitemap appear).

File: content/spin/v2/ai-sentiment-analysis-api-tutorial.md
Comment (mikkelhegn) [inline content/spin/v2/ai-sentiment-analysis-api-tutorial.md:289]: Awaiting a fix in the Rust SDK

File: content/spin/v2/key-value-store-tutorial.md
Comment (mikkelhegn) [inline content/spin/v2/key-value-store-tutorial.md:209]: @tpmccallum Do we need an update for this video?

File: content/spin/v2/key-value-store-tutorial.md
Comment (mikkelhegn) [inline content/spin/v2/key-value-store-tutorial.md:545]: @tpmccallum / @karthik2804 - Is this required to stay in this doc?

**Comment by itowlson** [comment]
Heads up that any `allowed_http_hosts = ["thing"]` are going to need to change to `allowed_outbound_hosts = ["https://thing"]`

File: content/spin/v2/key-value-store-tutorial.md
Comment (tpmccallum) [reply content/spin/v2/key-value-store-tutorial.md:209]: Yes, this will need to be remade to suite v2 if on the v2 page.
I have removed the v1 video content from this v2 page.

File: content/spin/v2/key-value-store-tutorial.md
Comment (tpmccallum) [reply content/spin/v2/key-value-store-tutorial.md:545]: Have removed this from the doc.

**Review by tpmccallum — APPROVED** [review]

File: content/spin/v2/ai-sentiment-analysis-api-tutorial.md
Comment (itowlson) [inline content/spin/v2/ai-sentiment-analysis-api-tutorial.md:264]: Is it intentional that you are no longer showing the surrounding fields?  The previous text said "note the positioning" and I'm not sure what it was aiming at or if that's still relevant.

File: content/spin/v2/ai-sentiment-analysis-api-tutorial.md
Comment (itowlson) [inline content/spin/v2/ai-sentiment-analysis-api-tutorial.md:289]: This appears to be merged now?

File: content/spin/v2/ai-sentiment-analysis-api-tutorial.md
Comment (itowlson) [inline content/spin/v2/ai-sentiment-analysis-api-tutorial.md:304]: Why does this now need `Serialize`?

File: content/spin/v2/ai-sentiment-analysis-api-tutorial.md
Comment (itowlson) [inline content/spin/v2/ai-sentiment-analysis-api-tutorial.md:339]: Does this compile?  I am bit surprised that `Request` is accepted for `Request<Json<SentimentAnalysisRequest>>` but if it works it works!

File: content/spin/v2/key-value-store-tutorial.md
Comment (itowlson) [inline content/spin/v2/key-value-store-tutorial.md:22]: Can't comment on next para but it links to a blog on Spin 0.9!

File: content/spin/v2/key-value-store-tutorial.md
Comment (itowlson) [inline content/spin/v2/key-value-store-tutorial.md:113]: Yeah this has not changed

File: content/spin/v2/key-value-store-tutorial.md
Comment (itowlson) [inline content/spin/v2/key-value-store-tutorial.md:116]: Can't comment on the file content sections but should these be manifest 2 format?

File: content/spin/v2/key-value-store-tutorial.md
Comment (itowlson) [inline content/spin/v2/key-value-store-tutorial.md:305]: I believe JS KV returns null for key not found rather than throwing (so this will need to be checked in the try block).  If I'm wrong we probably need to update our API guide

File: content/spin/v2/key-value-store-tutorial.md
Comment (itowlson) [inline content/spin/v2/key-value-store-tutorial.md:318]: Indent?

File: content/spin/v2/key-value-store-tutorial.md
Comment (itowlson) [inline content/spin/v2/key-value-store-tutorial.md:314]: The "if not .. else" can be hard for human readers to parse.  If we are adding an `else` then it would be good to flip this to "if exists { log found} else { log not found; status = 404 }"

File: content/spin/v2/key-value-store-tutorial.md
Comment (itowlson) [inline content/spin/v2/key-value-store-tutorial.md:503]: grammar
```suggestion
Great!,  The above command successfully returned our data as intended.
```

File: content/spin/v2/key-value-store-tutorial.md
Comment (itowlson) [inline content/spin/v2/key-value-store-tutorial.md:529]: I am not sure this paragraph is relevant at this stage.  Yes we are always happy for feedback but that is true of everything.  This API seems like it's past the "testing the water" stage.

File: content/spin/v2/serverless-ai-hello-world.md
Comment (itowlson) [inline content/spin/v2/serverless-ai-hello-world.md:151]:
```suggestion
allowed_outbound_hosts = []
```

File: content/spin/v2/serverless-ai-hello-world.md
Comment (itowlson) [inline content/spin/v2/serverless-ai-hello-world.md:211]: Can't leave a comment but do you want to update the Rust HTTP handler code?

**Review by itowlson — APPROVED** [review]
okay looking back I think most of these are nice to have rather than must so approved

**Comment by tpmccallum** [comment]
Hi @mikkelhegn 
Do you have time to build the tutorials in the pages you updated (pages that reside under the v2 tutorials menu)?

It will be ideal to know that these will successfully build and run for a user on 2.0. 

Also do you have time to replace the v1 manifest code blocks for the v2 pages that are linked to from the v2 tutorials menu?

Let me know; I will be starting early tomorrow and can respond then.

**Review by tpmccallum — CHANGES_REQUESTED** [review]
Hi @mikkelhegn 
Just a few notes, that we had better get fixed in the `/spin/v2/` path before we merge.

File: content/spin/v2/ai-sentiment-analysis-api-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/ai-sentiment-analysis-api-tutorial.md:289]: 👍🏻

File: content/spin/v2/ai-sentiment-analysis-api-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/ai-sentiment-analysis-api-tutorial.md:304]: It does not, might have been an intermediate step where this was needed while trying to get the new SDK working. Removing.

File: content/spin/v2/ai-sentiment-analysis-api-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/ai-sentiment-analysis-api-tutorial.md:339]: I'm also confused, but it compoles and works. See here: https://github.com/fermyon/spin/issues/1973

File: content/spin/v2/ai-sentiment-analysis-api-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/ai-sentiment-analysis-api-tutorial.md:264]: Yes, I didn't see a consistent implementation across all templates, in terms of what was in that section, so decided to so it this way. I don't think it decreases chance of success.

File: content/spin/v2/serverless-ai-hello-world.md
Comment (mikkelhegn) [reply content/spin/v2/serverless-ai-hello-world.md:211]: Looks like this is already taken care of. The code here compiles and works with rc1.

File: content/spin/v2/key-value-store-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/key-value-store-tutorial.md:529]: 👍🏻 - Removed.

File: content/spin/v2/key-value-store-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/key-value-store-tutorial.md:22]: I don't see that in here anymore?

File: content/spin/v2/key-value-store-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/key-value-store-tutorial.md:113]: Good - I see I misread this as "If one of your application's component cretes a new key/value STORE, ..."

File: content/spin/v2/key-value-store-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/key-value-store-tutorial.md:116]: Yes - this is an oversight from me

Comment (mikkelhegn) [reply content/spin/v2/key-value-store-tutorial.md:116]: Decided to combine to just show the commonality across templates

File: content/spin/v2/key-value-store-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/key-value-store-tutorial.md:305]: Fixed

File: content/spin/v2/key-value-store-tutorial.md
Comment (mikkelhegn) [reply content/spin/v2/key-value-store-tutorial.md:314]: 👍🏻

**Comment by mikkelhegn** [comment]
This is ready to go, but need an updated review from @tpmccallum

**Comment by mikkelhegn** [comment]
@tpmccallum - Looks like you updated the `package-lock.json` files in this PR. Was that intentional?
I'll leave it to you to merge this PR, once your good with it 👍🏻

**Comment by tpmccallum** [comment]
Hi @mikkelhegn 
Yeah, thanks, the package*.json stuff was a once off event (the dependabot made some PRs that we had to commit) sorry if this made the workflow harder. We all experienced the changes in our parked PRs.

**Comment by tpmccallum** [comment]
Hi @mikkelhegn 
Thanks so much for the update.

I just want to mention that there are still references to the v1 manifest in pages that come under the `v2` tutorial menu. Please see a list of line numbers below (of pages in the `/spinv2/` filepath) which mention the old v1 `[[component]]` format:

```
content/spin/v2/ai-sentiment-analysis-api-tutorial.md line 1014
content/spin/v2/ai-sentiment-analysis-api-tutorial.md line 1025
content/spin/v2/ai-sentiment-analysis-api-tutorial.md line 1032
content/spin/v2/key-value-store-tutorial.md line 112:
content/spin/v2/key-value-store-tutorial.md line 116
```

The following file has the `spin_manifest_version = “1”` in it also.

```
content/spin/v2/ai-sentiment-analysis-api-tutorial.md line 1007
```

**Comment by mikkelhegn** [comment]
Let me check

**Comment by mikkelhegn** [comment]
Ok - got it. I'll commit in 10 min.

**Comment by mikkelhegn** [comment]
@tpmccallum - Is now fixed.

**Review by tpmccallum — APPROVED** [review]
LGTM
Thanks so much @mikkelhegn

**Pull request merged**
